tracing-appender/src/rolling.rs

```diff
@@ -0,0 +1,128 @@
+use crate::inner::InnerAppender;
+use chrono::{DateTime, Datelike, TimeZone, Timelike, Utc};
+use std::fmt::Debug;
+use std::io;
+use std::path::Path;
+
+#[derive(Debug)]
+pub struct RollingFileAppender {
+    inner: InnerAppender,
+}
+
+impl RollingFileAppender {
+    pub fn new(
+        rotation: Rotation,
+        directory: impl AsRef<Path>,
+        file_name_prefix: impl AsRef<Path>,
+    ) -> RollingFileAppender {
+        RollingFileAppender {
+            inner: InnerAppender::new(
+                directory.as_ref(),
+                file_name_prefix.as_ref(),
+                rotation,
+                Utc::now(),
+            )
+            .expect("Failed to create appender"),
+        }
+    }
+}
+
+impl io::Write for RollingFileAppender {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        self.inner.write(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.inner.flush()
+    }
+}
+
+/// Creates an hourly, rolling file appender.
+///
+/// The appender returned by `rolling::hourly` can be used with `non_blocking` to create
+/// a non-blocking, hourly file appender.
+/// # Examples
+/// ``` rust
+///
+/// fn main() {
```

Comment:
Note that [clippy doesn't like `fn main()` in doctests](https://github.com/tokio-rs/tracing/pull/673/checks?check_run_id=586742212#step:4:495). Here, though, it _is_ serving a purpose, in that it shows the user that the guard should be held in `main`...this isn't super important to me, so I'm okay with removing the main, but I'd kinda prefer to put the `#[clippy::allow(needless_doctest_main)]` attribute on this function.